### PR TITLE
Refactor, regex bug fix, and additional functionality

### PR DESCRIPTION
**Pull request by MissMaximas**
Refactor: Split Go processes out into their own classes, then further broke them down so they will be more testable.

Regex Bug Fix: Fixed issue with `s/` showing up in the package names.

Additional Functionality: Failed tests will be highlighted. `all` can be placed into the packages to test, so it will test everything in a project, excluding the vendor.

### CODE REVIEW

**Comment by MissMaximas** [comment]
Needs tests.

File: go_processes/code_coverage.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        :param has_error: bool
+        :return: bool
+        """
+        err = False
+        output = ""
+        out, err = self._run_tests(base_package)
+
+        if err is not None and err is not "\n":
+            LOGGER.info(err)
+            has_error = True
```
Comment (jimah) [inline go_processes/code_coverage.py]: should this continue?

Comment (MissMaximas) [inline go_processes/code_coverage.py]: It was like this before, decided to keep the functionality the same and just fix it

**Comment by jimah** [comment]
This is also failing for me in Pipeline.

screen shot 2017-01-20 at 14 50 57

**Pull request merged**